src/models/legend.js

```diff
@@ -176,8 +177,22 @@ nv.models.legend = function() {
             if (align) {
 
                 var seriesWidths = [];
+                var largestKey = 0;
+
+                series.each(function(d,i) {
+                  if (getKey(d).length > largestKey) {
+                    largestKey = getKey(d).length;
+                  }
+                });
+
                 series.each(function(d,i) {
-                    var legendText = d3.select(this).select('text');
+                    var legendText;
+                    if (largestKey > maxKeyLength && getKey(d).length > maxKeyLength) { 
```

Comment:
Why do you calculate largestKey, shouldn't you just check if each one is larger than the maxKeyLength?  largestKey will always be larger if any of them are larger and you're doing both calculations so your'e basically doing the same thing twice...